pyccel/stdlib/cwrapper/cwrapper.c

```diff
@@ -0,0 +1,381 @@
+ *		a 	  : python array object
+ *      rank  : desired rank
+ * 	Returns		:
+ *		return true if no error occurred otherwise it will return false
+ *      and raise TypeError exception
+ * reference of the used c/python api function
+ * -------------------------------------------
+ * https://numpy.org/doc/stable/reference/c-api/array.html#c.PyArray_NDIM
+ */
+
+static bool _check_pyarray_rank(PyArrayObject *a, int rank)
+{
+	int	current_rank;
+
+	current_rank = PyArray_NDIM(a);
+	if (current_rank != rank)
+	{
+		PyErr_Format(PyExc_TypeError, "argument rank must be %d, not %d",
+			rank,
+			current_rank);
+		return false;
+	}
+
+	return true;
+}
+
+/*
+ * Function: _check_pyarray_order
+ * --------------------
+ * Check Python Object Order:
+ *
+ * 	Parameters	:
+ *		a 	  : python array object
+ *      flag  : desired order
+ * 	Returns		:
+ *		return true if no error occurred otherwise it will return false
+ *      and raise NotImplementedError exception
+ * reference of the used c/python api function
+ * -------------------------------------------
+ * https://numpy.org/doc/stable/reference/c-api/array.html#c.PyArray_CHKFLAGS
+ */
+
+static bool _check_pyarray_order(PyArrayObject *a, int flag)
+{
+	char	order;
+
+	if (flag == NO_ORDER_CHECK)
+		return true;
+
+	if (!PyArray_CHKFLAGS(a, flag))
+	{
+		order = flag == NPY_ARRAY_C_CONTIGUOUS ? 'C' : 'F';
+		PyErr_Format(PyExc_NotImplementedError,
+			"argument does not have the expected ordering (%c)", order);
+		return false;
+	}
+
+	return true;
+}
+
+/*
+ * Function: pyarray_to_ndarray
+ * ----------------------------
+ * A Cast function that convert numpy array variable into ndarray variable,
+ * by copying its information and data to a new variable of type ndarray struct
+ * and return this variable to be used inside c code.
+ * 	Parameters	:
+ *		o 	  : python array object
+ *      array : c ndarray object
+ *      dtype : desired data type enum
+ *		rank  : desired rank
+ *		flag  : desired order flag
+ * 	Returns		:
+ *		return true if no error occurred otherwise it will return false
+ *
+ * reference of the used c/numpy api function
+ * -------------------------------------------
+ * https://numpy.org/doc/stable/reference/c-api/array.html
+ */
+
+/*
+** convert numpy strides to nd_array strides, and return it in a new array, to
+** avoid the problem of different implementations of strides in numpy and ndarray.
+*/
+static int64_t	*_numpy_to_ndarray_strides(int64_t *np_strides, int type_size, int nd)
+{
+    int64_t *ndarray_strides;
+
+    ndarray_strides = (int64_t*)malloc(sizeof(int64_t) * nd);
+    for (int i = 0; i < nd; i++)
+        ndarray_strides[i] = np_strides[i] / type_size;
+
+    return ndarray_strides;
+}
+
+/*
+** copy numpy shape to nd_array shape, and return it in a new array, to
+** avoid the problem of variation of system architecture because numpy shape
+** is not saved in fixed length type.
+*/
+static int64_t     *_numpy_to_ndarray_shape(int64_t *np_shape, int nd)
```

Comment:
Can we use `npy_intp` for the numpy shape type? As we are already importing python things here it should be ok and it will fix issue #758

Reply:
yes